### PR TITLE
new synthtool: autocompressor

### PR DESCRIPTION
**Pull request by stanalbatross**
## About The Pull Request

Added a machine that automatically does cpr to the synthtool vendor

## Why It's Good For The Game

new synthtool good, can be used to help out synths when they don't have anyone to help them with CPR

## Changelog

:cl: Stan_Albatross, Nauticall
add: added the autocompressor, a new synthtool that automatically does CPR. Runs on an internal battery, can be attached to people just by clicking on them. Has a high risk of breaking bones if used alongside a person performing CPR. Sprites by Naut.
/:cl:

### CODE REVIEW

**Comment by JohnDoe50974** [comment]
Based

**Comment by MiniMeatwad** [comment]
Thought it said air compressor, and that you were just filling people's lungs with air.

**Review by Shad0vvs — CHANGES_REQUESTED** [review]
> tools/bootstrap/.cache/node-v14.16.1-x64/node.exe

Remove...

**Review by Shad0vvs — APPROVED** [review]

**Review by Shad0vvs — CHANGES_REQUESTED** [review]
revive_grace_period = 4820
It's too strong on corpses going PAST the revive cap.

**Comment by stanalbatross** [comment]
> revive_grace_period = 4820 It's too strong on corpses going PAST the revive cap.

yeah I did not intend for this to happen, gonna fix

**Comment by stanalbatross** [comment]
> revive_grace_period = 4820 It's too strong on corpses going PAST the revive cap.

we were just reading the vvs wrong, doing cpr also does this as revive_grace_period never goes down, it is only compared to timeofdeath

File: code/game/objects/items/tools/experimental_tools.dm [hunk 1]
```diff
@@ -127,3 +127,163 @@
...
+	update_icon()
+
+/obj/item/clothing/suit/auto_cpr/process()
+	if(!ishuman(loc))
+		end_cpr()
+		return PROCESS_KILL
+
+	var/mob/living/carbon/human/H = loc
+	if(H.wear_suit != src)
+		message_admins("killed process off non loc")
```
Comment (Shad0vvs) [inline code/game/objects/items/tools/experimental_tools.dm:252]:
```suggestion
```

File: code/game/objects/items/tools/experimental_tools.dm [hunk 2]
```diff
@@ -127,3 +127,163 @@
...
+
+	if(pdcell.charge <= 49)
+		src.visible_message(SPAN_NOTICE("\The [src] beeps its low power alarm."))
+		end_cpr()
+		return PROCESS_KILL
+
+	if(world.time > last_pump + 10 SECONDS)
+		last_pump = world.time
+		if(H.stat == UNCONSCIOUS)
+			var/suff = min(H.getOxyLoss(), 10) //Pre-merge level, less healing, more prevention of dieing.
```
Comment (Shad0vvs) [inline code/game/objects/items/tools/experimental_tools.dm:264]:
```suggestion
			var/suff = min(H.getOxyLoss(), 10) //Pre-merge level, less healing, more prevention of dying.
```

**Comment by Shad0vvs** [comment]
You wanted me to let you know to prevent putting it on perma people.

**Review by Shad0vvs — APPROVED** [review]

**Pull request merged**